builtins/amp-ad.js

```diff
@@ -108,8 +108,8 @@ export function installAd(win) {
        */
       this.unlistenViewportChanges_ = null;
 
-      /** @private {boolean} */
-      this.shouldSendIntersectionChanges_ = false;
+      /** @private {IntersectionObserver} */
+      this.intersectionObserver_ = null;
```

Comment:
Intersection observer calls the init which listens to messages from the iframe - so at that point we would need an iframe - does not make sense to set that late.